atom/browser/api/atom_api_web_contents.cc

```diff
@@ -884,26 +901,80 @@ void WebContents::DidGetRedirectForResourceRequest(
        details.headers.get());
 }
 
+void WebContents::DidStartNavigation(
+    content::NavigationHandle* navigation_handle) {
+  int frame_tree_node_id = navigation_handle->GetFrameTreeNodeId();
+  content::FrameTreeNode* frame_tree_node =
+    content::FrameTreeNode::GloballyFindByID(frame_tree_node_id);
+  content::RenderFrameHostManager* render_manager
+    = frame_tree_node->render_manager();
+  content::RenderFrameHostImpl* current_frame_host
+    = render_manager->current_frame_host();
+  content::RenderFrameHostImpl* pending_frame_host
+    = render_manager->pending_frame_host();
+  content::RenderFrameHostImpl* speculative_frame_host
+    = render_manager->speculative_frame_host();
+  content::RenderFrameHostImpl* frame_host =
+    speculative_frame_host ? speculative_frame_host : current_frame_host;
+
+  bool is_main_frame = navigation_handle->IsInMainFrame();
+  int frame_process_id = -1, frame_routing_id = -1;
+  if (frame_host) {
+    content::GlobalFrameRoutingId global_routing_id =
+      frame_host->GetGlobalFrameRoutingId();
+    frame_process_id = global_routing_id.child_id;
+    frame_routing_id = global_routing_id.frame_routing_id;
+  }
+  bool is_same_document = navigation_handle->IsSameDocument();
+  auto url = navigation_handle->GetURL();
+  Emit("did-start-navigation",
+       url,
+       is_same_document,
+       is_main_frame,
+       frame_process_id,
+       frame_routing_id);
+}
+
 void WebContents::DidFinishNavigation(
     content::NavigationHandle* navigation_handle) {
   bool is_main_frame = navigation_handle->IsInMainFrame();
+  content::RenderFrameHost* render_frame_host =
+    navigation_handle->GetRenderFrameHost();
```

Comment:
@zcbenz or @deepak1556 

it looks like this GetRenderFrameHost call is problematic (failing ci)

```
[6815:0418/031343.941870:FATAL:navigation_handle_impl.cc(318)] Check failed: state_ >= WILL_PROCESS_RESPONSE (1 vs. 6)This accessor should only be called after a response has been delivered for processing.
#0 0x7f833624db2d base::debug::StackTrace::StackTrace()
#1 0x7f833624bf5c base::debug::StackTrace::StackTrace()
#2 0x7f83362d2dba logging::LogMessage::~LogMessage()
#3 0x7f833ec55c96 content::NavigationHandleImpl::GetRenderFrameHost()
#4 0x000000c3e34b atom::api::WebContents::DidFinishNavigation()
```

Is this expected?  I can do the same thing here as in DidStartNavigation